docs/learn/introduction/04-the-domain-level.md

```diff
@@ -0,0 +1,351 @@
+active cells, as well as the number of cells per hardware device.
+
+By calling `ioDomainToVtk` method, we can also inspect the created domain (`grid`) via Paraview as the code generates a
+vtk
+file (`myDomain`). With an application working on a dense domain, the Paraview will show all the cells in the
+discretization box as active.
+Moreover, it will show the mapping between cells and hardware devices as
+reported in the following picture:
+
+![Mapping between cells and hardware devices](img/04-domain.vtk.png)
+
+<a name="field">
+### **Neon field**: defining data over the cartesian discretisation
+</a>
+
+Let's now allocate some metadata on top of the cartesian discretization we have just creates.
+Neon field are the tool for the task. Fields are allocated from a Neon grid.
+The type of a field depends on the metadata store on each cell.
+A field can allocate a number of components for each cell. This number of components
+which we call cardinality is defined when the field is created.
+
+```cpp linenums="72" title="Neon/tutorials/introduction/domainLevel/domainLevel.cpp"
+    //...
+    
+    // Step 3 -> Neon field: initializing a sphere through its signed distance function
+
+    // Creating a scalar field over the grid.
+    // Inactive cells will get associated with a default value of -100 */
+    auto sphereSdf = grid.newField<double>("sphereSdf",  // <- Given name of the field.
+                                           1,            // <- Number of field's component per grid point.
+                                           -100);        // <- Default value for non active points.
+
+    const double r = (n * voxelEdge / 2) * .3;
+
+    // Using the signed distance function of a sphere to initialize the field's values
+    // We leverage the forEachActiveCell method to easily iterate over the active cells.
+    sphereSdf.forEachActiveCell([&](const Neon::index_3d& idx, int, double& value) {
+        double sdf = sdfCenteredSphere(idx, dim, voxelEdge, r);
+        value = sdf;
+    });
+
+    // Exporting some information of the sdf on terminal and on a vtk file. 
+    NEON_INFO(sphereSdf.toString());
+    sphereSdf.ioToVtk("sdf", "sdf");
+    
+    return 0;
+}
+```
+
+![](img/04-initial-sdf.png)
+
+<a name="mapContainers">
+### **Neon map containers**: expanding the sphere via a level set
+</a>
+
+Let's now manipulate the sdf to expand zero level set of the sphere.
+The operation can be simply implemented by adding a constant to all the cells of the grid.
+The result will be a level set field (not a signed distance function).
+
+Until now the values of the `sphereSdf` field are stored on the host,
+indeed working on the host side is the only way we can do IO type of operations.
+The process of moving data to the devices is simple as it requires only a simple mehtod call.
+
+!!! Warning
+
+    All Neon operation related to runnig computation or moving data have a asynchronous semantic.
```

Comment:
[grammar] All Neon operations related to running computation or moving data have an asynchronous semantic.